packages/liveblocks-react/src/__tests__/useThreads.test.tsx

```diff
@@ -1063,6 +1071,193 @@ describe("useThreads", () => {
   });
 });
 
+describe("useThreads: error", () => {
+  beforeEach(() => {
+    jest.useFakeTimers();
+  });
+
+  afterEach(() => {
+    jest.clearAllTimers();
+    jest.useRealTimers(); // Restores the real timers
+  });
+
+  test("should retry with exponential backoff on error", async () => {
+    let getThreadsReqCount = 0;
+    server.use(
+      mockGetThreads((_req, res, ctx) => {
+        getThreadsReqCount++;
+        // Mock an error response from the server for the initial fetch
+        return res(ctx.status(500));
+      })
+    );
+
+    const {
+      roomCtx: { RoomProvider, useThreads },
+    } = createRoomContextForTest();
+
+    const { result, unmount } = renderHook(() => useThreads(), {
+      wrapper: ({ children }) => (
+        <RoomProvider id="room-id" initialPresence={{}}>
+          {children}
+        </RoomProvider>
+      ),
+    });
+
+    expect(result.current).toEqual({ isLoading: true });
+
+    // A new fetch request for the threads should have been made after the initial render
+    await waitFor(() => expect(getThreadsReqCount).toBe(1));
+
+    expect(result.current).toEqual({
+      threads: [],
+      isLoading: false,
+      error: expect.any(Error),
+    });
+
+    // The first retry should be made after ERROR_RETRY_INTERVAL * 2^0
+    jest.advanceTimersByTime(ERROR_RETRY_INTERVAL);
```

Comment:
Small pet peeve of mine: when writing tests, we should try to avoid importing constants and instead use literals or other static fixture data, to make the test cases easier to read and understand. (For actual code, this would be fine, of course!)

In this case, I'd suggest:

```suggestion
    jest.advanceTimersByTime(5 * SECONDS);
```

If ever we'd change the `ERROR_RETRY_INTERVAL`, I'd want this test case to fail.

Reply:
@nvie For my personal curiosity, about "If ever we'd change the `ERROR_RETRY_INTERVAL`, I'd want this test case to fail.": what would you do in that case, update the test? Isn't that the same result in the sense that this test will always be tied to the value used?

Reply:
Thanks for the link! I read the test as "it should do something after a specific amount of time, which is a known and specific amount, not sometime in the future" but yeah that perspective could lead to unforeseen consequences.

Reply:
Just read the article you shared. Really like this statement from the article. Thanks for sharing.

> Unit tests are meant to repeat values as a way to provide a second point-of-view on the correctness of a given piece of code